### PR TITLE
Changed docker-compose.yml to accomodate for apple silicon architecture

### PR DESCRIPTION
**Pull request by jacobrstone**
Reasons for fix:

- Quick start would not run on M2 machine after cloning, the mysql server would not start. 
- Mysql version updated from 5.7 -> 8.0 following suit with the [dev environment](https://github.com/opendialogai/opendialog-dev-environment)

In terminal: 

- `docker compose up --build`
- `docker-compose exec app bash docker/scripts/update-docker.sh` 

If the above doesn't work, do `docker compose down -v`, then repeat the above.

### CODE REVIEW

**Review by patshone-gsl — CHANGES_REQUESTED** [review]
It would good to have some information for our users about how to handle this upgrade in a way that doesn't force them to destroy their volume.

There's some narrative here - https://github.com/docker-library/mysql/pull/416 and further here - https://dev.mysql.com/doc/refman/8.0/en/docker-mysql-getting-started.html#docker-upgrading

Can we test the feasibility of providing an upgrade of the underlying data and then document

File: readme.md [hunk 1]
```diff
@@ -34,4 +34,21 @@ replacing dev with the tag you are interested in and then start it up again.
 
 ` docker-compose up -d --force-recreate app`
 
+## To upgrade your MySQL Server Version
```
Comment (patshone-gsl) [inline readme.md:37]: This is definitely getting there - I think the instructions would need to be more specific about how and when to change the mysql image version in docker-compose.yml

I'm assuming that in order to upgrade with the new mysql image version, we'd need to revert the version to 5.7 start the container, then stop it, upgrade the version to 8, run the upgrade scripts and then restart. Does that sound about right

Comment (jacobrstone) [reply readme.md:37]: I guess it also depends on when you change docker-compose.yml. If you've built the container with `image: mysql:5.7.27` then you can just upgrade. If you clone the repo, change the docker-compose.yml _first_ then run the commands, you wouldn't need to upgrade but of course you'd have no data. And then if for some reason you're already on 8.x, you'd be fine. 

I'm not sure what scenario you'd need to revert to a version of 5.7 though... 

Alternatively, we could have users backup their docker volumes and then used them after the wiped method is completed. Could be much easier/less verbose.

Comment (patshone-gsl) [reply readme.md]: Thanks Jacob, this is working well - the only thing I would add is a note about where to find the password for the root mysql user when running the command `docker exec -it quick-start-mysql-1 mysql_upgrade -uroot -p` and we're good to go

**Review by patshone-gsl — APPROVED** [review]
Thanks Jacob, this looks good

**Pull request merged**